phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -167,50 +180,152 @@ private void printHelpAndExit(Options options, int exitCode) {
             connection = ConnectionUtil.getInputConnection(configuration);
-            if (!isValidIndexTable(connection, qDataTable, indexTable)) {
-                throw new IllegalArgumentException(String.format(
-                    " %s is not an index table for %s ", qIndexTable, qDataTable));
+            long minDisableTimestamp = HConstants.LATEST_TIMESTAMP;
+            PTable indexWithMinDisableTimestamp = null;
+            
+            //Get Indexes in building state, minDisabledTimestamp 
+            List<String> disableIndexes = new ArrayList<String>();
+            List<PTable> disabledPIndexes = new ArrayList<PTable>();
+            for (PTable index : pdataTable.getIndexes()) {
+                if (index.getIndexState().equals(PIndexState.BUILDING)) {
+                    disableIndexes.add(index.getTableName().getString());
+                    disabledPIndexes.add(index);
+                    if (minDisableTimestamp > index.getIndexDisableTimestamp()) {
+                        minDisableTimestamp = index.getIndexDisableTimestamp();
+                        indexWithMinDisableTimestamp = index;
+                    }
+                }
+            }
+            
+            if (indexWithMinDisableTimestamp == null) {
+                throw new Exception("There is no index for a datatable to be rebuild:" + qDataTable);
             }
+            if (minDisableTimestamp == 0) {
+                throw new Exception("It seems Index " + indexWithMinDisableTimestamp
+                        + " has disable timestamp as 0 , please run IndexTool with IndexName to build it first");
+                // TODO probably we can initiate the job by ourself or can skip them while making the list for partial build with a warning
+            }
+            
+            long maxTimestamp = getMaxRebuildAsyncDate(schemaName, disableIndexes);
+            
+            //serialize index maintaienr in job conf with Base64 TODO: Need to find better way to serialize them in conf.
+            List<IndexMaintainer> maintainers = Lists.newArrayListWithExpectedSize(disabledPIndexes.size());
+            for (PTable index : disabledPIndexes) {
+                maintainers.add(index.getIndexMaintainer(pdataTable, connection.unwrap(PhoenixConnection.class)));
+            }
+            ImmutableBytesWritable indexMetaDataPtr = new ImmutableBytesWritable(ByteUtil.EMPTY_BYTE_ARRAY);
+            IndexMaintainer.serializeAdditional(pdataTable, indexMetaDataPtr, disabledPIndexes, connection.unwrap(PhoenixConnection.class));
+            PhoenixConfigurationUtil.setIndexMaintainers(configuration, indexMetaDataPtr);
+            
+            //Prepare raw scan 
+            Scan scan = IndexManagementUtil.newLocalStateScan(maintainers);
+            scan.setTimeRange(minDisableTimestamp - 1, maxTimestamp);
+            scan.setRaw(true);
+            scan.setCacheBlocks(false);
```

Comment:
Yes @chrajeshbabu , I did a below test and found the performance of both the scans(with or without cache) are same, file descriptor count during the test is almost same.

// with cache on
- create a new table
- Insert 500k records
- Flush the data
- scan with cache on

// with cache off
- create a new table again
- Insert 500k records
- Flush the data
- scan with cache off

Do you want me to do some other tests to verify the GC impact and all? or is it fine for now? as the cache is also kept off in org.apache.hadoop.hbase.mapreduce.Export by default.